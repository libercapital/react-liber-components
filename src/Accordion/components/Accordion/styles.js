import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
