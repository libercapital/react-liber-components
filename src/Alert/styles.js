import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
