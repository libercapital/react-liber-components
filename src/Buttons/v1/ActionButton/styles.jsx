import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
