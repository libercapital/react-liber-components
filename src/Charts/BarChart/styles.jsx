import styled from 'styled-components';
import { handleThemeFromObject } from 'Themes';

export const empty = null;

export const Slider = styled.input`
  width: ${props => (props.width ? `${props.width}px` : '100%')};
  appearance: none;
  height: 8.5px;
  background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  outline: none;
  border-radius: 4.25px;
  border: solid 2px ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};

  &::-webkit-slider-thumb {
    -webkit-appearance: none;
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-moz-range-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-ms-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &:focus {
    outline: none;
  }
  &::-moz-focus-outer {
    border: 0;
  }
  &::-moz-range-track {
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  }
`;

Slider.defaultProps = {
  type: 'range',
  valueAsNumber: true,
};
