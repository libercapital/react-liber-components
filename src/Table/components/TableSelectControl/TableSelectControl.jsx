import React from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  SelectedTitles,
  TotalSelectedTitles,
  Currency,
  DeselectIndicator,
} from './TableSelectControl.styles';

const renderSelectedTitlesLabel = (selectedTitles) => {
  if (!selectedTitles) {
    return 'Nenhum título selecionado';
  }
  if (selectedTitles === 1) {
    return `${selectedTitles} título selecionado`;
  }
  return `${selectedTitles} títulos selecionados`;
};

const TableSelectControl = (props) => {
  const {
    totalSelectedTitles,
    selectedTitles,
    onDeselect,
    type,
  } = props;
  return (
    <Container {...props}>
      <SelectedTitles type={type}>{renderSelectedTitlesLabel(selectedTitles)}</SelectedTitles>
      <TotalSelectedTitles type={type} onClick={onDeselect}>
        <Currency type={type}>R$</Currency>{totalSelectedTitles}
        {type !== 'empty' && <DeselectIndicator type={type} />}
      </TotalSelectedTitles>
    </Container>
  );
};

TableSelectControl.propTypes = {
  selectedTitles: PropTypes.number,
  totalSelectedTitles: PropTypes.string,
  type: PropTypes.string,
  onDeselect: PropTypes.func,
};

TableSelectControl.defaultProps = {
  selectedTitles: 0,
  totalSelectedTitles: '0,00',
  type: 'empty',
  onDeselect: () => {},
};

export default TableSelectControl;
