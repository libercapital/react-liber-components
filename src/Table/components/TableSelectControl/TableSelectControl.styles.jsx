import React from 'react';
import styled, { css } from 'styled-components';
import { MdClose } from '../../../Icons';
import { handleThemeFromObject as getTheme } from '../../../Themes';

export const Container = styled.div`
  width: 100%;
  max-height: 320px;
  display: flex;
  flex-direction: row;
`;

const selectedColors = (type = 'empty') => ({
  single: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.petroleum', '#4b6f85')};
    color: ${props => getTheme(props.theme, 'colors.v2.petroleum', '#4b6f85')};
    ${Container}:hover &{
      color: ${props => getTheme(props.theme, 'colors.v2.darkPetroleum', '#3C596B')};
      border-color: ${props => getTheme(props.theme, 'colors.v2.darkPetroleum', '#3C596B')};
    }
  `,
  multiple: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.yellow', '#fabe00')};
    color: ${props => getTheme(props.theme, 'colors.v2.yellow', '#fabe00')};
    ${Container}:hover &{
      color: ${props => getTheme(props.theme, 'colors.v2.darkYellow', '#F0B600')};
      border-color: ${props => getTheme(props.theme, 'colors.v2.darkYellow', '#F0B600')};
    }
  `,
  empty: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.white', '#ffffff')};
    color: ${props => getTheme(props.theme, 'colors.v2.grey3', '#c6d1d8')};
  `,
}[type]);

export const SelectedTitles = styled.div`
  ${props => selectedColors(props.type)}
  font-size: ${props => getTheme(props.theme, 'fontSizes.p', '16px;')};
  font-family: ${props => getTheme(props.theme, 'fontFamily', 'Roboto')};
  padding: 0 12px;
  border-radius: 4px 0px 0px 4px;
  height: 36px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  box-sizing: border-box;
  white-space: nowrap;
  transition: all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
`;

const totalSelectedColors = (type = 'empty') => ({
  single: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.petroleum', '#4b6f85')};
    color: ${props => getTheme(props.theme, 'colors.v2.white', '#ffffff')};
    background-color: ${props => getTheme(props.theme, 'colors.v2.petroleum', '#4b6f85')};
    ${Container}:hover &{
      background-color: ${props => getTheme(props.theme, 'colors.v2.darkPetroleum', '#3C596B')};
      border-color: ${props => getTheme(props.theme, 'colors.v2.darkPetroleum', '#3C596B')};
    }
  `,
  multiple: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.yellow', '#fabe00')};
    color: ${props => getTheme(props.theme, 'colors.v2.white', '#ffffff')};
    background-color: ${props => getTheme(props.theme, 'colors.v2.yellow', '#fabe00')};
    ${Container}:hover &{
      background-color: ${props => getTheme(props.theme, 'colors.v2.darkYellow', '#F0B600')};
      border-color: ${props => getTheme(props.theme, 'colors.v2.darkYellow', '#F0B600')};
    }
  `,
  empty: css`
    border: 1px solid ${props => getTheme(props.theme, 'colors.v2.grey5', '#ecf0f3')};
    color:  ${props => getTheme(props.theme, 'colors.v2.grey3', '#c6d1d8')};
    background-color: ${props => getTheme(props.theme, 'colors.v2.grey5', '#ecf0f3')};
    pointer-events: none;
  `,
}[type]);

export const TotalSelectedTitles = styled.button`
  ${props => totalSelectedColors(props.type)}
  font-size: ${props => getTheme(props.theme, 'fontSizes.p', '16px;')};
  font-family: ${props => getTheme(props.theme, 'fontFamily', 'Roboto')};
  font-weight: ${props => getTheme(props.theme, 'fontWeights.medium', 500)};
  padding: 0px 12px 0px 11px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  border-radius: 0px 4px 4px 0px;
  transition: all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
  white-space: nowrap;
`;

const colorsCurrency = (type = 'empty') => ({
  single: css`color: ${props => getTheme(props.theme, 'colors.v2.white', '#ffffff')};`,
  multiple: css`color: ${props => getTheme(props.theme, 'colors.v2.white', '#ffffff')};`,
  empty: css`color: ${props => getTheme(props.theme, 'colors.v2.grey3', '#c6d1d8')};`,
}[type]);

export const Currency = styled.div`
  ${props => colorsCurrency(props.type)}
  font-size: ${props => getTheme(props.theme, 'fontSizes.mobileMedium', '12px;')};
  font-family: ${props => getTheme(props.theme, 'fontFamily', 'Roboto')};
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-right: 4px;
`;

const deselectColors = (type = 'empty') => ({
  single: css`
    --x-color: ${props => getTheme(props.theme, 'colors.v2.petroleum', '#4b6f85')};
    ${Container}:hover &{
      --x-color: ${props => getTheme(props.theme, 'colors.v2.darkPetroleum', '#3C596B')};
    }
  `,
  multiple: css`
    --x-color: ${props => getTheme(props.theme, 'colors.v2.yellow', '#fabe00')};
    ${Container}:hover &{
      --x-color: ${props => getTheme(props.theme, 'colors.v2.darkYellow', '#F0B600')};
    }
  `,
  empty: css`--x-color: ${props => getTheme(props.theme, 'colors.v2.grey5', '#ecf0f3')};`,
}[type]);


const Close = styled(MdClose)`
  --size: 16px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: var(--x-color);
  transition: fill 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
`;

export const DeselectIndicator = styled.span`
  ${props => deselectColors(props.type)}
  border-radius: 4px;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.12);
  background-color: #ffffff;
  margin-left: 10px;
  border: 0px;
  width: 20px;
  height: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
`;


DeselectIndicator.defaultProps = {
  children: <Close />,
};
