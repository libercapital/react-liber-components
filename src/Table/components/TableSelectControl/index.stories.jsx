import React from 'react';
import styled from 'styled-components';
import { action } from '@storybook/addon-actions';
import TableSelectControl from './TableSelectControl';

const Container = styled.div`
  margin: 20px;
`;

const CustomizedSelectControl = styled(TableSelectControl)`
  width: auto;
`;

export default {
  title: 'Components 2.0|Table/Table Select Control',
};

export const emptyType = () => (
  <Container>
    <TableSelectControl
      selectedTitles={0}
      totalSelectedTitles="0,00"
      onDeselect={action('clicked')}
      type="empty"
    />
  </Container>
);

export const singleType = () => (
  <Container>
    <TableSelectControl
      selectedTitles={1}
      totalSelectedTitles="1.234,56"
      onDeselect={action('clicked')}
      type="single"
    />
  </Container>
);

export const multipleType = () => (
  <Container>
    <TableSelectControl
      selectedTitles={15}
      totalSelectedTitles="123.478,10"
      type="multiple"
      onDeselect={action('clicked')}
    />
  </Container>
);

export const customizedSelectControl = () => (
  <Container>
    <CustomizedSelectControl
      selectedTitles={15}
      totalSelectedTitles="123.478,10"
      type="multiple"
      onDeselect={action('clicked')}
    />
  </Container>
);
