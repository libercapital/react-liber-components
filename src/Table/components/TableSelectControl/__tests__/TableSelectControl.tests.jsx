import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { TotalSelectedTitles } from '../TableSelectControl.styles';
import TableSelectControl from '../TableSelectControl';


describe('Test for Action Button', () => {
  let props;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    props = {
      selectedTitles: 0,
      totalSelectedTitles: '0,00',
      onDeselect: sinon.spy(),
    };
  });
  it('should render correctly without props', () => {
    const wrapper = shallow(<TableSelectControl />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with props type=empty', () => {
    props.type = 'empty';
    const wrapper = shallow(<TableSelectControl {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with props type=single', () => {
    props.type = 'single';
    props.selectedTitles = 5;
    props.totalSelectedTitles = '100,00';
    const wrapper = shallow(<TableSelectControl {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with props type=multiple', () => {
    props.type = 'multiple';
    props.selectedTitles = 100;
    props.totalSelectedTitles = '10.000,00';
    const wrapper = shallow(<TableSelectControl {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call onDeselect on button click', () => {
    props.type = 'multiple';
    props.selectedTitles = 100;
    props.totalSelectedTitles = '10.000,00';
    const wrapper = shallow(<TableSelectControl {...props} />);
    const button = wrapper.find(TotalSelectedTitles);
    button.simulate('click');
    expect(props.onDeselect.called).toBeTruthy();
  });
});
