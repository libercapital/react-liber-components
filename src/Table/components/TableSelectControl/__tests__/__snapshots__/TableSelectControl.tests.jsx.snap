// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Action Button should render correctly with props type=empty 1`] = `
<TableSelectControlstyles__Container
  onDeselect={[Function]}
  selectedTitles={0}
  totalSelectedTitles="0,00"
  type="empty"
>
  <TableSelectControlstyles__SelectedTitles
    type="empty"
  >
    Nenhum título selecionado
  </TableSelectControlstyles__SelectedTitles>
  <TableSelectControlstyles__TotalSelectedTitles
    onClick={[Function]}
    type="empty"
  >
    <TableSelectControlstyles__Currency
      type="empty"
    >
      R$
    </TableSelectControlstyles__Currency>
    0,00
  </TableSelectControlstyles__TotalSelectedTitles>
</TableSelectControlstyles__Container>
`;

exports[`Test for Action Button should render correctly with props type=multiple 1`] = `
<TableSelectControlstyles__Container
  onDeselect={[Function]}
  selectedTitles={100}
  totalSelectedTitles="10.000,00"
  type="multiple"
>
  <TableSelectControlstyles__SelectedTitles
    type="multiple"
  >
    100 títulos selecionados
  </TableSelectControlstyles__SelectedTitles>
  <TableSelectControlstyles__TotalSelectedTitles
    onClick={[Function]}
    type="multiple"
  >
    <TableSelectControlstyles__Currency
      type="multiple"
    >
      R$
    </TableSelectControlstyles__Currency>
    10.000,00
    <TableSelectControlstyles__DeselectIndicator
      type="multiple"
    >
      <TableSelectControlstyles__Close />
    </TableSelectControlstyles__DeselectIndicator>
  </TableSelectControlstyles__TotalSelectedTitles>
</TableSelectControlstyles__Container>
`;

exports[`Test for Action Button should render correctly with props type=single 1`] = `
<TableSelectControlstyles__Container
  onDeselect={[Function]}
  selectedTitles={5}
  totalSelectedTitles="100,00"
  type="single"
>
  <TableSelectControlstyles__SelectedTitles
    type="single"
  >
    5 títulos selecionados
  </TableSelectControlstyles__SelectedTitles>
  <TableSelectControlstyles__TotalSelectedTitles
    onClick={[Function]}
    type="single"
  >
    <TableSelectControlstyles__Currency
      type="single"
    >
      R$
    </TableSelectControlstyles__Currency>
    100,00
    <TableSelectControlstyles__DeselectIndicator
      type="single"
    >
      <TableSelectControlstyles__Close />
    </TableSelectControlstyles__DeselectIndicator>
  </TableSelectControlstyles__TotalSelectedTitles>
</TableSelectControlstyles__Container>
`;

exports[`Test for Action Button should render correctly without props 1`] = `
<TableSelectControlstyles__Container
  onDeselect={[Function]}
  selectedTitles={0}
  totalSelectedTitles="0,00"
  type="empty"
>
  <TableSelectControlstyles__SelectedTitles
    type="empty"
  >
    Nenhum título selecionado
  </TableSelectControlstyles__SelectedTitles>
  <TableSelectControlstyles__TotalSelectedTitles
    onClick={[Function]}
    type="empty"
  >
    <TableSelectControlstyles__Currency
      type="empty"
    >
      R$
    </TableSelectControlstyles__Currency>
    0,00
  </TableSelectControlstyles__TotalSelectedTitles>
</TableSelectControlstyles__Container>
`;
